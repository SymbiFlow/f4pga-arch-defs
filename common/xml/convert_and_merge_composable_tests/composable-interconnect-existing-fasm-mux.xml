<?xml version="1.0"?>
<!-- Test the special handling around the fasm_mux metadata doesn't destroy
     existing values.
     -->
<xml>
 <pb_type name="parent">
  <input name="i0" />
  <input name="i1" />
  <output name="o" />

  <pb_type name="child">
   <input name="i" />
   <output name="o" />
  </pb_type>

  <interconnect>
   <direct name="CIN_TO_CARRY0" input="COMMON_SLICE.CIN" output="CARRY4_VPR.CIN" >
    <pack_pattern name="CARRYCHAIN"/>
    <metadata>
     <meta name="fasm_mux">
      COMMON_SLICE.CIN = PRECYINIT.CIN
     </meta>
    </metadata>
   </direct>
   <mux name="DFFMUX"
    input="CARRY4_VPR.O3 CARRY4_VPR.CO_FABRIC3 COMMON_SLICE.DO6 COMMON_SLICE.DO5 COMMON_SLICE.DX"
    output="SLICE_FF.D[3]" >
    <metadata>
     <meta name="fasm_mux">
      COMMON_SLICE.DX = DFFMUX.DX
      COMMON_SLICE.DO5 = DFFMUX.O5
      COMMON_SLICE.DO6 = DFFMUX.O6
      CARRY4_VPR.CO_FABRIC3 = DFFMUX.CY
      CARRY4_VPR.O3 = DFFMUX.XOR
     </meta>
    </metadata>
   </mux>
   <mux name="CARRY_DI0" input="COMMON_SLICE.AO5 COMMON_SLICE.AX" output="CARRY4_VPR.DI0" >
    <delay_constant max=".105e-9" in_port="COMMON_SLICE.AX" out_port="CARRY4_VPR.DI0" />
    <metadata>
     <meta name="fasm_mux">
      COMMON_SLICE.AO5 = CARRY4.ACY0
      COMMON_SLICE.AX = NULL
     </meta>
    </metadata>
   </mux>
  </interconnect>
 </pb_type>
</xml>
