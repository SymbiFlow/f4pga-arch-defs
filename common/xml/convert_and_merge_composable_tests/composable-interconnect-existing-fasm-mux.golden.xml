<?xml version="1.0"?>
<xml>
  <pb_type name="parent">
    <input name="i0"/>
    <input name="i1"/>
    <output name="o"/>
    <pb_type name="child">
      <input name="i"/>
      <output name="o"/>
    </pb_type>
    <interconnect>
      <direct input="COMMON_SLICE.CIN" name="CIN_TO_CARRY0" output="CARRY4_VPR.CIN">
        <pack_pattern in_port="COMMON_SLICE.CIN" name="CARRYCHAIN" out_port="CARRY4_VPR.CIN"/>
        <metadata>
          <meta name="fasm_mux">
      COMMON_SLICE.CIN = PRECYINIT.CIN
     </meta>
        </metadata>
      </direct>
      <mux input="COMMON_SLICE.AO5 COMMON_SLICE.AX" name="CARRY_DI0" output="CARRY4_VPR.DI0">
        <metadata>
          <meta name="fasm_mux">
      COMMON_SLICE.AO5 = CARRY4.ACY0
      COMMON_SLICE.AX = NULL
     </meta>
        </metadata>
        <delay_constant in_port="COMMON_SLICE.AX" max=".105e-9" out_port="CARRY4_VPR.DI0"/>
      </mux>
      <mux input="CARRY4_VPR.O3 CARRY4_VPR.CO_FABRIC3 COMMON_SLICE.DO6 COMMON_SLICE.DO5 COMMON_SLICE.DX" name="DFFMUX" output="SLICE_FF.D[3]">
        <metadata>
          <meta name="fasm_mux">
      COMMON_SLICE.DX = DFFMUX.DX
      COMMON_SLICE.DO5 = DFFMUX.O5
      COMMON_SLICE.DO6 = DFFMUX.O6
      CARRY4_VPR.CO_FABRIC3 = DFFMUX.CY
      CARRY4_VPR.O3 = DFFMUX.XOR
     </meta>
        </metadata>
      </mux>
    </interconnect>
  </pb_type>
</xml>
